lib/src/main/kotlin/com/swmansion/starknet/data/serializers/JsonRpcTransactionTracePolymorphicSerializer.kt

```diff
@@ -6,15 +6,25 @@ import kotlinx.serialization.json.*
 
 internal object JsonRpcTransactionTracePolymorphicSerializer :
     JsonContentPolymorphicSerializer<TransactionTrace>(TransactionTrace::class) {
+    private fun selectInvokeTransactionTraceDeserializer(jsonObject: JsonObject): DeserializationStrategy<out InvokeTransactionTrace> {
+        val executeInvocation = jsonObject["execute_invocation"]?.jsonObject ?: throw IllegalStateException("execute_invocation is missing.")
```

Comment:
let's improve on the exception messaging, to better point out what went wrong

Reply:
What other details do you have in mind? 

The trace should already contain the info that:
1. `JsonRpcTransactionTracePolymorphicSerializer` failed.
2. Specifically, `selectInvokeTransactionTraceDeserializer` failed.
3. Specifically, because `execute_invocation` field is missing.

Reply:
`"execute_invocation is missing."`

let's add at least that's from simulate transaction endpoint; possibly we could print out the response we got and state it may be node's fault, that should help identify the real problem

Reply:
Well technically the trace could be used elsewhere. And the stack trace would already contain info that `simulateTransactions` endpoint was used.

But I agree we could add info indicating that it may be the node's fault, I'm just not sure we ever print out the full response, or in that case, jsonObject when using serializers.

Reply:
> Well technically the serializer could be used elsewhere.

do you have any specific place in mind? it's hard for me to imagine where someone would use trace transaction serializer, apart from trace transactions 🤔 

>  I'm just not sure we ever print out the full response, or in that case, jsonObject when using serializers.

sure, that was just an idea, not a requirement :+1: we should stick to conventions we have; if we do not do it anywhere, let's stick to it. Still, the message could be improved imo

Reply:
> I meant other endpoints that are not simulateTransactions. We currently don't support those, but it could be used in starknet_traceTransaction or starknet_traceBlockTransactions endpoints.

Ah ok, so futureproofing. Sure, that makes sense then.

> I will add the info that the error might be caused by the node-specific implementation and response of simulateTransactions. We are unlikely to add any other trace api endpoints to SDK before 0.5.0 anyway?

let's keep the messaging simple, and just state that node sent over a wrong response (maybe let's not mention any node-specific implementations? my point is - we actually don't know, the potential error users encounters may be not connected to node implementation at all).

Yes, probably not before 0.5.0. Maybe after it is released, we'll see :+1: